compiler/shared/main/scala/mlscript/compiler/ir/IR.scala

```diff
@@ -137,7 +138,8 @@ enum Node:
   case Case(scrut: Name, cases: Ls[(ClassInfo, Node)])
   // Intermediate forms:
   case LetExpr(name: Name, expr: Expr, body: Node)
-  case LetCall(names: Ls[Name], defn: DefnRef, args: Ls[TrivialExpr], body: Node)
+  case LetCall(names: Ls[Name], defn: DefnRef, args: Ls[TrivialExpr], isTailRec: Bool, body: Node)
+  case AssignField(assignee: Name, clsInfo: ClassInfo, fieldName: Str, value: TrivialExpr, body: Node)
```

Comment:
Why is this a `Node` and not an `Expr`? In principle, nodes are for control operations and `Expr`s are for expression-like things like this.

Reply:
I don't see why `AssignField` should be an `Expr`, since that would make using it quite clunky, i.e. you would need `let dummy = (assign trivialExpr to foo.bar)`.

Reply:
What's the problem with this? It seems more regular than having an entirely different node type with its own body, requiring more code to deal with node shapes.

Reply:
resolved in latest commit